internal.sh

```diff
@@ -0,0 +1,86 @@
+
+rm -fr crc-cluster-kube-apiserver-operator
+rm -fr crc-cluster-kube-controller-manager-operator
+
+function run_preflight_checks() {
+        if [ -z "${OPENSHIFT_PULL_SECRET_PATH-}" ]; then
+            echo "OpenShift pull secret file path must be specified through the OPENSHIFT_PULL_SECRET_PATH environment variable"
+            exit 1
+        elif [ ! -f ${OPENSHIFT_PULL_SECRET_PATH} ]; then
+            echo "Provided OPENSHIFT_PULL_SECRET_PATH (${OPENSHIFT_PULL_SECRET_PATH}) does not exists"
+            exit 1
+        fi
+}
+
+run_preflight_checks
+
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp}
+
+# If user defined the OPENSHIFT_VERSION environment variable then use it.
+if test -n "${OPENSHIFT_VERSION-}"; then
+    OPENSHIFT_RELEASE_VERSION=${OPENSHIFT_VERSION}
+    echo "Using release ${OPENSHIFT_RELEASE_VERSION} from OPENSHIFT_VERSION"
+else
+    OPENSHIFT_RELEASE_VERSION="$(curl -L "${MIRROR}"/candidate-4.11/release.txt | sed -n 's/^ *Version: *//p')"
+    if test -n "${OPENSHIFT_RELEASE_VERSION}"; then
+        echo "Using release ${OPENSHIFT_RELEASE_VERSION} from the latest mirror"
+    else
+        echo "Unable to determine an OpenShift release version.  You may want to set the OPENSHIFT_VERSION environment variable explicitly."
+        exit 1
+    fi
+fi
+
+if test -z "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE-}"; then
+    OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$(curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/release.txt" | sed -n 's/^Pull From: //p')"
+elif test -n "${OPENSHIFT_VERSION-}"; then
+    echo "Both OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE and OPENSHIFT_VERSION are set, OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE will take precedence"
+    echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: $OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"
+    echo "OPENSHIFT_VERSION: $OPENSHIFT_VERSION"
+fi
+echo "Setting OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+
+mkdir -p openshift-clients/linux
+curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux-${OPENSHIFT_RELEASE_VERSION}.tar.gz" | tar -zx -C openshift-clients/linux oc
+OC=./openshift-clients/linux/oc
+
+function patch_and_push_image() {
+    local image_name=$1
+    image=$(${OC} adm release info -a ${OPENSHIFT_PULL_SECRET_PATH} ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} --image-for=${image_name})
+    vcs_ref=$(${OC} image info -a ${OPENSHIFT_PULL_SECRET_PATH} ${image} -ojson | jq -r '.config.config.Labels."vcs-ref"')
+    url=$(${OC} image info -a ${OPENSHIFT_PULL_SECRET_PATH} ${image} -ojson | jq -r '.config.config.Labels.url')
+    url_tag=$(basename ${url})
+    rhpkg clone containers/crc-${image_name}
+    pushd crc-${image_name}
+    git remote add upstream git://pkgs.devel.redhat.com/containers/ose-${image_name}
+    git fetch --all
+    git checkout --track  origin/rhaos-4.11-rhel-8
+    git merge ${vcs_ref}
+    git push -f origin HEAD
```

Comment:
Why force-push?

Reply:
@cfergeau dist-git need force push for changes, without that it fails.

Reply:
One very important part of building on brew/dist-git is that given a build, it is easy to find the corresponding source code/git history (through the `vcs-ref` tag). force pushing breaks this, which is bad. It's also not clear why it's needed from the steps I can see in this PR.

Reply:
There was some post hook which was failing, will check with `4.11.4` and comment here those failure.